### ISSUE TITLE
use optimizer Momentum with learning_rate=fluid.layers.piecewise_decay

### ISSUE THREAD

**Issue by seiriosPlus** [PaddlePaddle/Paddle#11144]
use optimizer Momentum with learning_rate=fluid.layers.piecewise_decay get this exception:

```
Traceback (most recent call last):
  File "/models/image_classification/se_resnext_high_api.py", line 278, in <module>
    train()
  File "/models/image_classification/se_resnext_high_api.py", line 227, in train
    param_path="/workspace/models")
  File "/usr/local/lib/python2.7/dist-packages/paddle/fluid/trainer.py", line 126, in __init__
    optimize_ops, params_grads = optimizer.minimize(loss)
  File "/usr/local/lib/python2.7/dist-packages/paddle/fluid/optimizer.py", line 253, in minimize
    startup_program)
  File "/usr/local/lib/python2.7/dist-packages/paddle/fluid/optimizer.py", line 212, in create_optimization_pass
    self._create_global_learning_rate()
  File "/usr/local/lib/python2.7/dist-packages/paddle/fluid/optimizer.py", line 73, in _create_global_learning_rate
    "learning rate variable is created outside optimizer,"
TypeError: learning rate variable is created outside optimizer, cannot create new learning rate variable for the new program
```
PaddlePaddle: latest with new API

**Comment by seiriosPlus** [comment]
related #11157

**Comment by sidgoyal78** [comment]
Fixed with #11168 and #11172
